README.md

```diff
@@ -2,31 +2,435 @@
 
 ## Learning Goals
 
-- Learning Goal 1
-- Learning Goal 2
+- Discuss message passing in object-oriented programming
+- Introduce **UML Sequence Diagrams** for visualizing object interactions
+- Implement object interactions in Java
 
 ## Introduction
 
-A 1-2 sentence summary of what will be covered.
+A Java program often consists of multiple related objects
+that collaborate to implement the desired application functionality.
+The collaboration can be viewed as "message passing", where one object
+sends a message asking another object to do something.  In Java
+we implement message passing through method invocation.
 
-## Topic 1
+## Code Along
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+Fork and clone this lesson.  We will build an application
+that demonstrates interactions among associated objects:
 
-## Topic 2
+- a `Motorcycle` object is associated with two `Tire` objects
+- a `Car` object is associated with four `Tire` objects
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+- The `Main` class contains a `main` method that creates an instance of `Motorcycle`
+  and implicitly calls the `toString` and explicitly calls the`rideThroughMud` methods on the object.
+- The `Motorcycle` class has fields for the `make` and `model`, along
+  with  fields `frontTire` and `rearTire` that reference instances of the `Tire` class.
+- The `Car` class has fields for the `make` and `model`, along
+  with an array named `tires` that stores references to 4 `Tire` objects.
+- The `Tire` class has instance variables `airPressure` and `clean`,
+  along with class constants `MIN_RECOMMENDED_PRESSURE` and `MIN_RECOMMENDED_PRESSURE`.
```

Comment:
I think one of these should be `MAX_RECOMMENDED_PRESSURE`

Reply:
fixed